src/main/java/io/opentracing/contrib/tracerresolver/TracerResolver.java

```diff
@@ -55,6 +55,14 @@
      * @return The resolved Tracer or {@code null} if none was resolved.
      */
     public static Tracer resolveTracer() {
+        try { // Take care NOT to import GlobalTracer as it is an optional dependency and may not be on the classpath.
+            if (io.opentracing.util.GlobalTracer.isRegistered()) {
+                return logResolved(io.opentracing.util.GlobalTracer.get());
+            }
+        } catch (NoClassDefFoundError globalTracerNotInClasspath) {
```

Comment:
I'm unclear how this works - doesn't referring to the class explicitly creates a link-time dependency that will cause `NoClassDefFoundError` as soon as you try to load `TracerResolver.java`?